test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+        authorizationId
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_ACCOUNT_IS_NOT_WHITELISTED);
+    });
+
+    it("Is reverted if the payment authorization ID is zero", async () => {
+      await expect(cardPaymentProcessor.connect(admin).clearPayment(
+        ZERO_BYTES16_STRING
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_AUTHORIZATION_ID_IS_ZERO);
+    });
+
+    it("Is reverted if the payment with the provided authorization ID does not exist", async () => {
+      await expect(cardPaymentProcessor.connect(admin).clearPayment(
+        createBytesString(payment.authorizationId + 1, BYTES16_LENGTH)
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_DOES_NOT_EXIST);
+    });
+
+    it("Does not transfer tokens, emits the correct event, changes the state properly", async () => {
+      await checkCardPaymentProcessorState([payment]);
+      const expectedClearedBalance: number = payment.amount;
+      const expectedUnclearedBalance: number = 0;
+      await expect(cardPaymentProcessor.connect(admin).clearPayment(
+        authorizationId
+      )).to.changeTokenBalances(
+        brlcMock,
+        [cardPaymentProcessor, payment.account],
+        [0, 0]
+      ).and.to.emit(
+        cardPaymentProcessor,
+        "ClearPayment"
+      ).withArgs(
+        authorizationId,
+        payment.account.address,
+        payment.amount,
+        expectedClearedBalance,
+        expectedUnclearedBalance,
+        payment.revocationCounter || 0
+      );
+      payment.status = PaymentStatus.Cleared;
+      await checkCardPaymentProcessorState([payment]);
+    });
+
+    it("Is reverted if the payment has already been cleared", async () => {
+      await proveTx(cardPaymentProcessor.connect(admin).clearPayment(
+        authorizationId
+      ));
+      await expect(cardPaymentProcessor.connect(admin).clearPayment(
+        authorizationId
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_IS_CLEARED);
+    });
+  });
+
+  describe("Function 'clearPayments()'", async () => {
+    let payments: TestPayment[];
+    let admin: SignerWithAddress;
+    let authorizationIds: string[];
+    let accountAddresses: string[];
+    let expectedClearedBalances: number[];
+    let expectedUnclearedBalances: number[];
+
+    beforeEach(async () => {
+      payments = [
+        {
+          authorizationId: 123,
+          account: user1,
+          amount: 234,
+          status: PaymentStatus.Nonexistent,
+          makingPaymentCorrelationId: 345,
+        },
+        {
+          authorizationId: 456,
+          account: user2,
+          amount: 567,
+          status: PaymentStatus.Nonexistent,
+          makingPaymentCorrelationId: 789,
+        }
+      ]
+      admin = user2;
+      authorizationIds = payments.map(payment => createBytesString(payment.authorizationId, BYTES16_LENGTH));
+      accountAddresses = payments.map(payment => payment.account.address);
+      expectedClearedBalances = payments.map(payment => payment.amount);
```

Comment:
Codacy has a fix for the issue: [Replace `payment` with `(payment)`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
      expectedClearedBalances = payments.map((payment) => payment.amount);
```